### PR TITLE
Lock shard session heartbeat

### PR DESCRIPTION
**Pull request by harshit-gangal**
This is to keep the connection alive on tablet to keep holding the lock that the current session hold.
Heartbeat will get triggered only if the session is active i.e. it receives a query for execution and also based on when the last time heartbeat was sent to tablet (currently it is 5 sec). 
This is put under a flag `lock_heartbeat_time`

### CODE REVIEW

**Review by systay — APPROVED** [review]

File: go/vt/vtgate/safe_session.go [hunk 1]
```diff
@@ -370,6 +371,22 @@ func (session *SafeSession) SetLockSession(lockSession *vtgatepb.Session_ShardSe
 	session.mu.Lock()
 	defer session.mu.Unlock()
 	session.LockSession = lockSession
+	session.LastLockHeartbeat = time.Now().Unix()
+}
+
+//UpdateLockHeartbeat updates the LastLockHeartbeat time
```
Comment (sougou) [inline go/vt/vtgate/safe_session.go:377]: nit: our coding standards require a space between `//` and the first word :).

File: go/vt/vtgate/scatter_conn.go [hunk 2]
```diff
@@ -166,6 +168,21 @@ func (stc *ScatterConn) ExecuteMultiShard(
 	var mu sync.Mutex
 	qr = new(sqltypes.Result)
 
+	if session.InLockSession() && session.TriggerLockHeartBeat() {
+		go func() {
```
Comment (sougou) [inline go/vt/vtgate/scatter_conn.go:172]: This is better than what we have now, but won't cover all use cases. For example, a bulk insert that takes longer than 5s will expire the lock.

I think the only reliable method would be to run an independent ticker that keeps sending heartbeats. But that is much more complicated than this. So, this is a good compromise.

File: go/vt/vtgate/safe_session.go (hunk 1 above)
Comment (systay) [reply go/vt/vtgate/safe_session.go:377]: We should teach the linter so we don't forget

File: go/vt/vtgate/scatter_conn.go (hunk 2 above)
Comment (harshit-gangal) [reply go/vt/vtgate/scatter_conn.go:172]: I will create a future task for it.

**Pull request merged**